### ISSUE TITLE
Support non-string Map keys

### ISSUE THREAD

**Issue by kaikoga** [ArpEngine/ArpDomain#45]
Should be done after #31 (and probably after #44)

- [ ] `IMap<Int, SomeValue>`
- [ ] `IMap<Int, SomeReference>`
- [ ] `IMap<IArpObject, SomeValue>`
- [ ] `IMap<IArpObject, SomeReference>`
- [ ] `IMap<Enum, SomeValue>`
- [ ] `IMap<Enum, SomeReference>`

`Bool` map keys are technically possible but low priority.
`Float` map keys... I think they are invalid
